test/integration/pipeline/batch_pipeline_test.cpp

```diff
@@ -336,3 +362,103 @@ INSTANTIATE_TEST_CASE_P(BatchPipelineParameterizedTest,
                         // https://github.com/google/googletest/issues/1419
                         Values(interface::types::BatchType::ATOMIC,
                                interface::types::BatchType::ORDERED), );
+
+/**
+ * Test that a batch would not be passed to stateful validation when one
+ * transaction is fully signed and another is not signed at all.
+ * @given
+ *   two users - the first user has quorum two
+ *   an atomic batch with two transactions
+ *   the first transaction is signed with only one valid signature
+ *   the second transaciton is not signed
+ *   the batch is sent to Iroha and in a pending state
+ * @when
+ *   the first transaction acquires the second valid signature
+ *   and the batch is sent again to Iroha
+ * @then
+ *   the batch is sill in a pending state
+ *   because the second transaction has no signatures
+ */
+TEST_F(BatchPipelineTest, SemisignedAtomicBatch) {
+  auto batch = framework::batch::makeTestBatch(
+      prepareTransferAssetBuilder(
+          kFirstUserId, kSecondUserId, kAssetA, "1.0", 2),
+      prepareTransferAssetBuilder(kSecondUserId, kFirstUserId, kAssetB, "1.0"));
+
+  batch = addSignaturesFromKeyPairs(batch, 0, kFirstUserKeypair);
+  auto firstTxHash = batch->transactions()[0]->hash();
+
+  integration_framework::IntegrationTestFramework itf(2);
+  itf.setInitialState(kAdminKeypair)
+      .sendTxAwait(createFirstUser(), CHECK_TXS_QUANTITY(1))
+      .sendTxAwait(createSecondUser(), CHECK_TXS_QUANTITY(1))
+      .sendTxAwait(createRole(), CHECK_TXS_QUANTITY(1))
+      .sendTxAwait(addRoleToUsers(), CHECK_TXS_QUANTITY(1))
+      .sendTxAwait(
+          createAndAddAssets(kFirstUserId, kAssetA, "10.0", kFirstUserKeypair),
+          CHECK_TXS_QUANTITY(1))
+      .sendTxAwait(createAndAddAssets(
+          kSecondUserId, kAssetB, "20.0", kSecondUserKeypair))
```

Comment:
Similar block of initial transactions is repeated in every test case. It would be nice of you to put it into some sort of initialization function.